src/app.ts

```diff
@@ -0,0 +1,69 @@
+import bodyParser from 'body-parser';
+import compression from 'compression';
+import cookieParser from 'cookie-parser';
+import config from 'config';
+import cors from 'cors';
+import express from 'express';
+
+import { Controller } from "./interfaces";
+import { documentValidationMiddleware } from './middlewares/document-validation.middleware';
+import { requestBodyValidationMiddleware } from './middlewares/request-body-validation.middleware';
+import { problemMiddleware } from './middlewares/problem.middleware'
+import { logger } from './utils/logger';
+
+export class App {
+  private app: express.Application;
+  private port: string | number;
+  private env: string;
+
+  constructor(controller: Controller[]) {
+    this.app = express();
+    this.port = process.env.PORT || 3000;
+    this.env = process.env.NODE_ENV || 'development';
+
+    this.initializeMiddlewares();
+    // initialize validation middlewares
+    this.initializeValidation();
+    // initialize all controllers
+    this.initializeControllers(controller);
+    // initialize error handlings
+    this.initializeErrorHandling();
+  }
+
+  public listen() {
+    this.app.listen(this.port, () => {
+      logger.info(`=================================`);
+      logger.info(`= ENV: ${this.env}`);
+      logger.info(`= 🚀 AsyncAPI Server API listening on the port ${this.port}`);
+      logger.info(`=================================`);
+    });
+  }
+
+  public getServer() {
+    return this.app;
+  }
+
+  private initializeMiddlewares() {
+    this.app.use(cors({ origin: config.get('cors.origin'), credentials: config.get('cors.credentials') }));
```

Comment:
How CORS and allowed origins are gonna be handled here? Are we allowing all by default?

Reply:
Maybe I'm answering here https://github.com/asyncapi/server-api/pull/7/files#r765582927

Reply:
Answered https://github.com/asyncapi/server-api/pull/7#discussion_r765667336 :)